src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -879,6 +889,14 @@ public async Task<int> GetCaretPositionAsync(CancellationToken cancellationToken
             return bufferPosition.Position;
         }
 
+        public async Task GoToDefinitionAsync(CancellationToken cancellationToken)
+        {
+            await TestServices.Shell.ExecuteCommandAsync(VSConstants.VSStd97CmdID.GotoDefn, cancellationToken);
+            await TestServices.Workspace.WaitForAllAsyncOperationsAsync(
+                new[] { FeatureAttribute.Workspace, FeatureAttribute.NavigateTo },
+                cancellationToken);
```

Comment:
Why the workspace wait _after_ the command but not before?

Reply:
I was accounting for the possibility that navigation may be asynchronous. It may be possible to remove this; I'm not sure.

Reply:
Specifically, this was my new replacement for the following old code:

https://github.com/dotnet/roslyn/blob/ce1461d5fb325f10ad0213cfba5bfbcb295a2676/src/VisualStudio/IntegrationTest/TestUtilities/OutOfProcess/Editor_OutOfProc.cs#L328-L331